core/src/main/scala/tamer/Registry.scala

```diff
@@ -15,22 +18,28 @@ object Registry {
   val live: URLayer[Has[SchemaRegistryClient], Has[Registry]] = ZLayer.fromService { client =>
     new Registry {
       private[this] final val logTask: Task[LogWriter[Task]] = log4sFromName.provide("tamer.registry")
+      private[this] final def getId(subject: String, schema: ParsedSchema, log: LogWriter[Task]): Task[Int] =
+        Task(client.getId(subject, schema)).tap(id => log.debug(s"retrieved existing writer schema id: $id"))
+      private[this] final def register(subject: String, schema: ParsedSchema, log: LogWriter[Task]): Task[Int] =
+        Task(client.register(subject, schema)).tap(id => log.info(s"registered with id $id new subject $subject writer schema $schema"))
+      private[this] final def get(id: Int, log: LogWriter[Task]): Task[ParsedSchema] =
+        Task(client.getSchemaById(id)).tap(_ => log.debug(s"retrieved writer schema id: $id"))
+      @nowarn private[this] final def verify(schema: ParsedSchema, writerSchema: ParsedSchema): Task[Unit] =
+        Task(schema.isBackwardCompatible(writerSchema).asScala)
+          .filterOrElse(_.isEmpty) { errors =>
+            ZIO.fail(TamerError(s"backwards incompatible schema: ${errors.mkString(", ")}"))
+          }
+          .unit
 
-      override final def getOrRegisterId(subject: String, schema: ParsedSchema): Task[Int] =
-        for {
-          s   <- UIO.succeed(schema)
-          log <- logTask
-          id <-
-            Task(client.getId(subject, s)).tap(id => log.debug(s"retrieved existing writer schema id: $id")) <>
-              Task(client.register(subject, s)).tap(id => log.info(s"registered with id $id new subject $subject writer schema $schema"))
-        } yield id
-      override final def verifySchema(id: Int, schema: ParsedSchema): Task[Unit] =
-        for {
-          s            <- UIO.succeed(schema)
-          log          <- logTask
-          writerSchema <- Task(client.getSchemaById(id)).tap(_ => log.debug(s"retrieved writer schema id: $id"))
-          _            <- Task(s.isBackwardCompatible(writerSchema)).tapError(t => log.error(s"schema supplied cannot read payload: ${t.getLocalizedMessage}"))
```

Comment:
turns out this wasn't very useful, since that API doesn't throw if not backwards compatible but returns a non empty list of strings, representing all errors, see [this](https://github.com/laserdisc-io/tamer/pull/390/files#diff-20c26aa90f0f810f60916628e933f724aee86922819874a1005a48b8a5c54d66R29) for the fix